crates/flux-middle/src/rty/mod.rs

```diff
@@ -157,10 +157,12 @@ pub enum Sort {
     Tuple(List<Sort>),
     Func(FuncSort),
     App(SortCtor, List<Sort>),
+    Var(usize),
 }
 
 #[derive(Clone, PartialEq, Eq, Hash, TyEncodable, TyDecodable)]
 pub struct FuncSort {
+    params: usize,
```

Comment:
I think it's a good idea to add something like `Binder`, but it doesn't have to be as general, we can have it just for function sorts. Maybe

```rust
pub struct PolyFuncSort {
    params: usize,
    fsort: FuncSort
}
```

and then 

```rust
pub enum Sort {
    ...
    Func(PolyFuncSort),
    ...
}
```

I'd also make `Sort::subst` private and only expose it through `PolyFuncSort`, maybe:

```rust
impl PolyFuncSort {
    pub fn instantiate(&self,  args: &[Sort]) -> FuncSort {
        let inputs_and_output = func_sort
            .inputs_and_output
            .iter()
            .map(|sort| sort.subst(&args))
            .collect();
        fhir::FuncSort { inputs_and_output }
    }
}
```

We will probably also need a `skip_binder`.